src/analytics/redux/tracked_events.js

```diff
@@ -64,20 +71,61 @@ function trackBlocksHandler(
 	} );
 }
 
+/**
+ * Helper function to track if a block is being dragged.
+ *
+ * @param {boolean} isDragging A boolean value to track if a block is being dragged
+ * @return {void}
+ */
+function setIsDraggingBlock( isDragging ) {
+	isDraggingBlock = isDragging;
+}
+
 /**
  * Helper function to track block movement events.
  *
- * @param {string} action Type of movement (arrows/drag & drop)
+ * @param {string[]} clientIds The client IDs of the blocks being moved
+ * @param {string}   action    Type of movement (arrows/drag & drop)
  * @return {void}
  */
-function trackBlockMoved( action ) {
+function trackBlockMoved( clientIds, action ) {
+	const block = select( 'core/block-editor' ).getBlocksByClientId(
+		clientIds?.[ 0 ]
+	)?.[ 0 ];
```

Comment:
If we only want to retrieve the data of the first block, we might consider using [the `getBlock` selector](https://github.com/WordPress/gutenberg/blob/9e1ad7c15c43c7d4aa7257cf6fdeaad15431bc74/packages/block-editor/src/store/selectors.js#L124-L152) and pass the first index of `clientIds`. In fact, that selector is the one used within the `getBlocksByClientId` selector ([reference](https://github.com/WordPress/gutenberg/blob/9e1ad7c15c43c7d4aa7257cf6fdeaad15431bc74/packages/block-editor/src/store/selectors.js#L333-L336)).

Reply:
Thanks for the suggestion! I updated in https://github.com/wordpress-mobile/gutenberg-mobile/pull/4833/commits/c8b88ce25a9f738d52f5c2e955983fb0c42415e1